invenio_records_resources/records/systemfields/index.py

```diff
@@ -29,10 +30,16 @@ def __init__(self, index_or_alias, search_alias=None):
             self._index = Index(index_or_alias, using=current_search_client)
         # Set search alias name directly on the index
         self._index.search_alias = search_alias or self._index._name
+        self._prefix_set = False
 
     #
     # Data descriptor methods (i.e. attribute access)
     #
     def __get__(self, record, owner=None):
         """Get the persistent identifier."""
+        if not self._prefix_set:
```

Comment:
This is not ideal, but we cannot access `current_app` in the constructor since it's instantiated before an application context is pushed. Aternatively, I could make the whole `self._index` a `LocalProxy`... Both feel hackish.